packages/@aws-cdk/custom-resources/lib/aws-custom-resource/aws-custom-resource.ts

```diff
@@ -308,6 +309,23 @@ export interface AwsCustomResourceProps {
    * ID for the function's name. For more information, see Name Type.
    */
   readonly functionName?: string;
+
+  /**
+   * The vpc to provision the lambda function in.
+   *
+   * @default - the function is not provisioned inside a vpc.
+   */
+  readonly vpc?: ec2.IVpc;
+
+  /**
+   * Which subnets from the VPC to place the lambda function in.
+   *
+   * Only used if 'vpc' is supplied. Note: internet access for Lambdas
```

Comment:
The comments here are doing a lot of heavy lifting. Rather than commenting that this can only be supplied if VPC is supplied, I think we need the contract/code to enforce this. Based off this I would be unclear if vpcSubnets is also required when vpc is supplied.

Reply:
I think like a lot of situations with `lambda.Function` this may be tricky? Because I think `vpc.selectSubnets()`/`vpc.selectSubnets(undefined)` will return private subnets if you only have private subnets and may return other values if that's not the case. It's a little hard to validate since `ec2.SubnetSelection` can have such a variety of options. The base construct seems to be pretty good at catching obviously-broken scenarios.

So I _think_ the comments are just documenting some of the synth-time failures that can occur if the expected behavior isn't happening, even though those checks happen in the Lambda package and not here.

Reply:
Right, there's validation happening elsewhere that will catch some of this. Given that, there should be test cases to ensure that it fails how we expect it to fail in those scenarios.